job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/DslFactory/workflowMultiBranchJob.groovy

```diff
@@ -0,0 +1,17 @@
+workflowMultiBranchJob('example') {
+    sources {
+        branchSource {
+            id('exampleId')
+            remote('')
+            credentialsId('')
+            includes('')
+            excludes('')
+            ignoreOnPushNotifications(false)
+        }
+    }
+    orphanedItemStrategy {
```

Comment:
There is an extension point for the orphaned item strategy and there can be more than one strategy in the future, so we need to distinguish them in the DSL. Add a `defaultStrategy` context to make room for other strategies:

```groovy
orphanedItemStrategy {
  defaultStrategy {
    pruneDeadBranches(true)
    daysToKeep(0)
    numToKeep(0)
  }
}
```

I don't like `defaultStrategy`, but `default` is a key word, so we can't use that. Better alternatives are welcome.